src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -57,18 +60,29 @@ public static FormulaValue FromJson(JsonElement element)
                     return new BooleanValue(IRContext.NotInSource(FormulaType.Boolean), true);
 
                 case JsonValueKind.Object:
-                    return RecordFromJsonObject(element);
+                    return RecordFromJsonObject(element, formulaType as RecordType);
 
                 case JsonValueKind.Array:
-                    return TableFromJsonArray(element);
+                    if (formulaType == null || formulaType is TableType)
+                    {
+                        return TableFromJsonArray(element, formulaType as TableType);
+                    }
+                    else if (formulaType is UntypedObjectType uot)
```

Comment:
We should return array of UntypedObjectType

Reply:
element is an array and it will be serialized properly